apps/system/js/app_window.js

```diff
@@ -169,6 +154,52 @@
     } else if (this.rearWindow) {
       this.rearWindow.setFrontWindow(this);
     }
+
+    // W3C web app manifest "display" property takes precedence
+    if (this.manifest && this.manifest.display) {
```

Comment:
Argh, all these cases are starting to get messy, especially with the 2+ fallbacks. I wonder if we can get on a migration plan here to drop the fallbacks in 3.0?